### ISSUE TITLE
Password mode

### ISSUE THREAD

**Issue by jackpot51** [MovingtoMars/liner#25]
I would like to use liner to capture usernames, followed by passwords. Currently the change required would be to not echo characters as they are received, or perhaps to use a mask char.

A mask char may be easier to implement, as it would not invalidate any of the editing interfaces

**Comment by MovingtoMars** [comment]
Does password mode belong in the readline implementation? To me, it seems like the features offered by a line editor aren't used when entering a password. 

I would think password entering would be in the program itself, eg `sudo`.

**Comment by iamcodemaker** [comment]
This should probably be handled by the readline library. Otherwise the user will have to manually disable the echoing of characters and handle escape codes and stuff themselves (this is the whole point of a readline lib). Shouldn't be too difficult to implement, just need to disable the echoing of characters for password input.

**Comment by iamcodemaker** [comment]
So there is actually a `getpass()` function in libc which is obsolete. It is recommend to use the termios functions to manually disable echoing and stuch. There is an `rpassword` crate that does just that.

**Comment by iamcodemaker** [comment]
I tested using liner with rpassword to read passwords and it works as expected. That's probably the best route here.

**Comment by MovingtoMars** [comment]
Shouldn't the program that wants a password just use `rpassword` directly?

**Comment by iamcodemaker** [comment]
Yes, that's how the example code I linked works. Initially I thought it belonged in liner, but now I'm thinking just using rpassword directly would work better.

**Comment by iamcodemaker** [comment]
`Termion` also has a [read_passwd](https://docs.rs/termion/1.5.1/termion/input/trait.TermRead.html#method.read_passwd) method if you wanted to avoid additional dependencies.
